File: JavaScript_Basic_04_2023_SolutionsOfProblems/05.Ex_PB-JS-While-Loop/P02.ExamPreparation.js
function examPreparation(input) {
    let badGradeNumber = Number(input[0]);
    let badGradeCounter = 0, sumGrade = 0, i = 1;
    while (input[i] !== "Enough" && badGradeCounter < badGradeNumber) {
        i++;
        currentGrade = Number(input[i]);
        if (currentGrade <= 4) {
            badGradeCounter++;
        }
        sumGrade += currentGrade;
        i++;
    }
    if (badGradeCounter < badGradeNumber) {
        console.log(`Average score: ${(sumGrade / ((i - 1) / 2)).toFixed(2)}`);
        console.log(`Number of problems: ${(i - 1) / 2}`);
        console.log(`Last problem: ${input[i - 2]}`);
    } else {
        console.log(`You need a break, ${badGradeCounter} poor grades.`);
    }
}
examPreparation(["2",
    "Income",
    "3",
    "Game Info",
    "6",
    "Best Player",
    "4"])

File: Java_Basic_Jan_2023_SolutionsOfProblems/P05Ex_While_Loop/P02_ExamPreparation.java
package P05Ex_While_Loop;

import java.util.Scanner;

public class P02_ExamPreparation {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int maxBadResults = Integer.parseInt(scanner.nextLine());
        String nameProblem = "", lastProblem = "";
        int result = 0, badResults = 0, sum = 0, count = 0;
        nameProblem = scanner.nextLine();
        while (!nameProblem.equals("Enough")) {
            lastProblem = nameProblem;
            result = Integer.parseInt(scanner.nextLine());
            sum += result;
            count++;
            if (result <= 4) {
                badResults++;
            }
            if (badResults >= maxBadResults) {
                break;
            }
            nameProblem = scanner.nextLine();
        }
        if (nameProblem.equals("Enough")) {
            System.out.printf("Average score: %.2f%n", sum * 1.0 / count);
            System.out.printf("Number of problems: %d%n", count);
            System.out.printf("Last problem: %s", lastProblem);
        } else {
            System.out.printf("You need a break, %d poor grades.", badResults);
        }
    }
}
